python/dask_cudf/dask_cudf/core.py

```diff
@@ -332,11 +332,28 @@ def repartition(self, *args, **kwargs):
         return super().repartition(*args, **kwargs)
 
     @_dask_cudf_nvtx_annotate
-    def shuffle(self, *args, shuffle=None, **kwargs):
+    def shuffle(self, index, shuffle=None, **kwargs):
         """Wraps dask.dataframe DataFrame.shuffle method"""
-        return super().shuffle(
-            *args, shuffle=_get_shuffle_type(shuffle), **kwargs
-        )
+
+        shuffle = _get_shuffle_type(shuffle)
+        if shuffle == "explicit-comms" and isinstance(index, (str, list)):
+            # NOTE: The upstream shuffle does not directly support
+            # the "explicit-comms" option, so it will create a
+            # new "_partitions" column before calling down to
+            # `rearrange_by_column`. The logic below avoids the
+            # creation of a temporary "_partitions" column
+            from dask.dataframe.shuffle import rearrange_by_column
+
+            index = [index] if isinstance(index, str) else list(index)
+            if set(index).issubset(self.columns):
+                return rearrange_by_column(
+                    self,
+                    index,
+                    shuffle=shuffle,
+                    **kwargs,
+                )
+
+        return super().shuffle(index, shuffle=shuffle, **kwargs)
```

Comment:
Unfortunately, there was a subtle bug in #11576 - We cannot pass `shuffle="explicit-comms"` to the upstream shuffle aglorithm, or we will end up trying to create (and drop) the "_partitions" column twice.

This seems to be the only reasonable fix (besides supporting "explicit-comms" directly in `dask.dataframe`)

Reply:
I proposed something here: https://github.com/dask/dask/pull/9521 that would square this circle, I think.

Reply:
>I proposed something here: https://github.com/dask/dask/pull/9521 that would square this circle, I think.

I agree that we should go in that direction, but will probably need to use a workaround like this until there is an upstream consensus.